Makefile

```diff
@@ -0,0 +1,167 @@
+.ONESHELL:
+SHELL := /bin/bash
+MODERNCVDIR =.
+MANUALDIR = $(MODERNCVDIR)/manual
+
+# version and date of the current release. This gets updated upon calling
+# either the rule version or the rule release
+VERSION = v2.1.0-40-gfe4d968-dirty
+VERSIONDATE = 2021/01/25
+# user callable NEW option, to specify the new version. If unspecified, the
+# new version gets determined by git.
+ifdef NEW
+    VERSIONNEXT = $(NEW)
+else
+    VERSIONNEXT = $(shell git describe --tags --dirty)
+endif
+VERSIONDATENEXT = $(shell date +"%Y\/%m\/%d")
+TARBALL=moderncv-$(VERSIONNEXT).tar
+
+EXAMPLESDIR = examples
+# MANUALTEX is used by the userguide method that operates in the 
+# $(MANUALDIR) folder. MANUAL is used by methods operating in 
+# the $(MODERNCVDIR) folder. This is to ensuer that the userguide 
+# is also buildable within the manual folder.
+MANUALTEX = moderncv_userguide.tex
+MANUAL = $(MANUALDIR)/$(MANUALTEX)
+TEMPLATE = $(MODERNCVDIR)/template.tex
+TEMPLATEBIB = publications.bib
+TEMPLATEBASE = $(basename $(TEMPLATE))
+TEMPLATEPDF = $(addsuffix .pdf,$(TEMPLATEBASE))
+MANUAL_BASE = $(basename $(MANUAL))
+MANUALPDF = $(addsuffix .pdf,$(MANUAL_BASE))
+
+# redefine the template rule depending on whether the user has specified STYLE
+# or not.
+ifdef STYLE
+	# in this case user has specified STYLE
+else
+	STYLE = casual
+endif
+template: $(TEMPLATE) $(TEMPLATEBIB)
+	if [[ $(strip $(STYLE)) == "casual" ]]; then
+		# build template in default style
+		latexmk -pdflua -bibtex -quiet "$<"
+	else
+		# build template in style $(STYLE). This assumes that casual is the default.
+		sedstring="s/moderncvstyle{casual}/moderncvstyle{$(STYLE)}/g"
+		sed -i $$sedstring $(TEMPLATE)
+		# build template in specified style
+		latexmk -pdflua -bibtex -quiet "$<"
+		# reset template to default value
+		sedstring="s/moderncvstyle{$(STYLE)}/moderncvstyle{casual}/g"
+		sed -i $$sedstring $(TEMPLATE)
+	fi
+
+
+templates: $(TEMPLATE) $(TEMPLATEBIB)
+	# build the template for each style and store pdfs in the examples folder
+	# this is done to include these expamples in release tar balls.
+	mkdir -p $(EXAMPLESDIR)
+	previousstyle="casual"
+	for style in casual classic banking oldstyle fancy; do
+		sedstring="s/moderncvstyle{$$previousstyle}/moderncvstyle{$$style}/g"
+		sed -i $$sedstring $(TEMPLATE)
+		latexmk -pdflua -bibtex -quiet $(TEMPLATE)
+		cp $(TEMPLATEPDF) $(EXAMPLESDIR)/$(TEMPLATEBASE)-$$style.pdf
+		mv $(TEMPLATEPDF) $(MANUALDIR)/$(TEMPLATEBASE)-$$style.pdf
+		previousstyle=$$style
+		unset sedstring
+	done
+	sedstring="s/moderncvstyle{$$previousstyle}/moderncvstyle{casual}/g"
+	sed -i $$sedstring $(TEMPLATE)
+
+
+userguide: templates $(MANUAL)
+	# build the user guide. Since the guide includes the template examples, we
+	# build those first by calling the templates rule.
+	cd $(MANUALDIR)
+	./format_files_for_userguide.py 
+	lualatex $(MANUALTEX)
+	lualatex $(MANUALTEX)
```

Comment:
Why not latexmk?

Reply:
I have had issues in the past with latexmk, therefor I avoid it.